tree/dataframe/inc/ROOT/RDF/RLoopManager.hxx

```diff
@@ -95,6 +95,15 @@ class RLoopManager : public RNodeBase {
       }
    };
 
+   /// A RAII object that calls RLoopManager::CleanUpTask at destruction
+   struct RCallCleanUpTask {
```

Comment:
Consider moving the class definition into the source file, keeping just the friend declaration - I guess this class isn't anybody's business?

Reply:
ah i always forget that friend declarations "count" as forward decls

Reply:
it gets a bit annoying with namespaces: https://godbolt.org/z/8ad1vxzvr , but ok, probably better than having it as part of the class body